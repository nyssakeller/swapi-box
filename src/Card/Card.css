@import url('https://fonts.googleapis.com/css?family=Orbitron');

article {
  background-color: rgba(32, 32, 32, 0.7);
  display: inline-block;
  height: 350px;
  margin: 20px;
  width: 300px;
}

.card p {
  color: white;
  font-size: 18px;
}

.card h2 {
  color: #fbde33;
  font-family: Orbitron;
  font-size: 30px;
}

.favorite-button {
  bottom: 25px;
  height: 35px;
  position: relative;
  left: 125px;
  margin-bottom: 0;
  width: 35px;
}