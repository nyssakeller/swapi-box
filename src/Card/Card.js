import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';


const Card = ({name, description, type, number, favoriteStatus, favoriteCard, residents, dataObj}) => {
  const toggleClass = favoriteStatus ? 'card favorite' : 'card';

  return (
    <article className={toggleClass}>
      <button 
        className='favorite-button'
        onClick={() => favoriteCard(dataObj)}>
      </button>

      <h2>{name}</h2>
      <p>{description}</p>
      <p>{type}</p>
      <p>{number}</p>
      <p>{residents}</p>
    </article>
  );  
};

export default Card;

Card.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  type: PropTypes.string,
  number: PropTypes.string,
  residents: PropTypes.string,
  dataObj: PropTypes.object,
  favoriteCard: PropTypes.func
};