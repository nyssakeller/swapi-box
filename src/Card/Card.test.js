import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {

  it('match snapshot', () => {
    const wrapper = shallow(<Card />)
    expect(wrapper).toMatchSnapshot();
  })

})