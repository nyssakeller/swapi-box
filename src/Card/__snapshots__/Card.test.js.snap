// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-button"
          onClick={[Function]}
/>,
        <h2 />,
        <p />,
        <p />,
        <p />,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-button"
            onClick={[Function]}
/>,
          <h2 />,
          <p />,
          <p />,
          <p />,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
